src/github/client/github-client.ts

```diff
@@ -192,4 +190,14 @@ export default class GitHubClient {
 		return response?.data?.data;
 	}
 
-}
+	/**
+	 * Get the repositiry contents from PATH
+	 */
+	public async getRepositoryContent(owner: string, repo: string, path: string) : Promise<AxiosResponse<Octokit.GitGetBlobResponse>>  {
```

Comment:
From the GitHub docs it looks more like `Octokit.ReposGetContentsResponse` is the expected return type here. That also contains the size of the file so we can throw an error in the service of the size is bigger than x.